src/System.Private.CoreLib/shared/System/String.Searching.cs

```diff
@@ -19,7 +19,7 @@ public bool Contains(string value)
             return SpanHelpers.IndexOf(
                 ref _firstChar,
                 Length,
-                ref value._firstChar,
+                ref value!._firstChar,      // TODO-NULLABLE: Compiler Bug?
```

Comment:
CC. @jaredpar, @gafter. Is this expected? `#nullable enable` is set for this file (and in the file that contains `_firstChar`: https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/src/System/String.CoreCLR.cs#L26) and the `value` parameter is non-nullable, so I wouldn't expect it to complain here.

Reply:
@cston

Reply:
This is not a bug as written. The test above actually makes us think that the argument may be null, and because we don't know that the ThrowArgumentNullException helper never returns, we think the code below is reachable. We probably need an attribute to indicate that a method will never return.

Reply:
> because we don't know that the ThrowArgumentNullException helper never returns

Ah, right. I completely missed that bit.

> We probably need an attribute to indicate that a method will never return.

We have at least a dozen issues on csharplang proposing this and similar, so it should be easy to find one to champion 😄

Reply:
(I might have slight preference towards my own proposal, however: https://github.com/dotnet/csharplang/issues/739)

Reply:
> We probably need an attribute to indicate that a method will never return.

https://github.com/dotnet/roslyn/issues/127
https://github.com/dotnet/roslyn/issues/59
https://github.com/dotnet/roslyn/issues/1226
https://github.com/dotnet/csharplang/issues/538